### PR TITLE
fix: the issue reported in 1780.

### PR DESCRIPTION
**Pull request by astanfel**
Setting the privilege to usage when we see IMPORTED PRIVILEGES for a database grant when checking the current permissions on the database. The provider just needs to check to see if the role in the resource has an existing usage grant on the database, if not it will issue the IMPORTED PRIVILEGES database grant again. 

## Test Plan

1. Define a snowflake_database_grant resource in Terraform to grant IMPORTED PRIVILEGES to a Snowflake role on a database that is created from a Data Share.
1. Apply the Terraform configuration to create the grant in Snowflake.
1. Remove the IMPORTED PRIVILEGES grant from the database outside of Terraform.
1. Run terraform plan.

* [ ] …

## References

* https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1780

### CODE REVIEW

**Comment by astanfel** [comment]
fixes https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1780

**Comment by astanfel** [comment]
@scottwinkler is there a way that I can test a "feature version" of my provider updates prior to merging this change?

File: docs/resources/email_notification_integration.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
```
Comment (astanfel) [inline docs/resources/email_notification_integration.md:1]: Looks like this just wasn't caught from a previous change, I just made it when I ran make docs

**Comment by astanfel** [comment]
testing 

```bash 
┌─(~/Documents/SoFi/terraform/imported_privs_testing)──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(tstanfel@SOFI-C02FJ423MD6R:s013)─┐
└─(12:03:29)──> tfp                                                                                                                                                                                                                            ──(Tue,May23)─┘
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - snowflake-labs/snowflake in /Users/tstanfel/go/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
module.genesys_a3s_data_share.snowflake_database.this: Refreshing state... [id=GENESYS_A3S_DATA_SHARE]
module.genesys_a3s_data_share.snowflake_role.read: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ]
module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|]
module.genesys_a3s_data_share.snowflake_role_grants.read["true"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ|DATA_CONSUMER_PUBLIC_READ_ONLY|]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
┌─(~/Documents/SoFi/terraform/imported_privs_testing)──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(tstanfel@SOFI-C02FJ423MD6R:s013)─┐
└─(12:03:40)──> tfp                                                                                                                                                                                                                            ──(Tue,May23)─┘
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - snowflake-labs/snowflake in /Users/tstanfel/go/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
module.genesys_a3s_data_share.snowflake_database.this: Refreshing state... [id=GENESYS_A3S_DATA_SHARE]
module.genesys_a3s_data_share.snowflake_role.read: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ]
module.genesys_a3s_data_share.snowflake_role_grants.read["true"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ|DATA_CONSUMER_PUBLIC_READ_ONLY|]
module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"] will be updated in-place
  ~ resource "snowflake_database_grant" "read" {
        id                     = "GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|"
      ~ roles                  = [
          + "GENESYS_A3S_DATA_SHARE__READ",
        ]
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
┌─(~/Documents/SoFi/terraform/imported_privs_testing)──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(tstanfel@SOFI-C02FJ423MD6R:s013)─┐
└─(12:04:02)──> tfa                                                                                                                                                                                                                            ──(Tue,May23)─┘
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - snowflake-labs/snowflake in /Users/tstanfel/go/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
module.genesys_a3s_data_share.snowflake_database.this: Refreshing state... [id=GENESYS_A3S_DATA_SHARE]
module.genesys_a3s_data_share.snowflake_role.read: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ]
module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|]
module.genesys_a3s_data_share.snowflake_role_grants.read["true"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ|DATA_CONSUMER_PUBLIC_READ_ONLY|]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"] will be updated in-place
  ~ resource "snowflake_database_grant" "read" {
        id                     = "GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|"
      ~ roles                  = [
          + "GENESYS_A3S_DATA_SHARE__READ",
        ]
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"]: Modifying... [id=GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|]
module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"]: Modifications complete after 1s [id=GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
┌─(~/Documents/SoFi/terraform/imported_privs_testing)──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(tstanfel@SOFI-C02FJ423MD6R:s013)─┐
└─(12:04:14)──> tfp                                                                                                                                                                                                                            ──(Tue,May23)─┘
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI configuration:
│  - snowflake-labs/snowflake in /Users/tstanfel/go/bin
│
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
module.genesys_a3s_data_share.snowflake_database.this: Refreshing state... [id=GENESYS_A3S_DATA_SHARE]
module.genesys_a3s_data_share.snowflake_role.read: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ]
module.genesys_a3s_data_share.snowflake_database_grant.read["IMPORTED PRIVILEGES"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE|IMPORTED PRIVILEGES|false|GENESYS_A3S_DATA_SHARE__READ|]
module.genesys_a3s_data_share.snowflake_role_grants.read["true"]: Refreshing state... [id=GENESYS_A3S_DATA_SHARE__READ|DATA_CONSUMER_PUBLIC_READ_ONLY|]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
┌─(~/Documents/SoFi/terraform/imported_privs_testing)──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(tstanfel@SOFI-C02FJ423MD6R:s013)─┐
└─(12:04:50)──>

```

File: pkg/resources/database_grant.go [hunk 2]
```diff
@@ -142,6 +145,14 @@ func ReadDatabaseGrant(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error reading database grant: %w", err)
 	}
 
+	// Then set it back to imported privledges for Terraform to execute the grant.
+	if privilege == "IMPORTED PRIVILEGES" {
+		priv := d.Set("privilege", "IMPORTED PRIVILEGES")
+		if priv := := d.Set("privilege", "USAGE"); priv != nil {
```
Comment (astanfel) [inline pkg/resources/database_grant.go:151]:
```suggestion
		if priv := d.Set("privilege", "USAGE"); priv != nil {
```

**Review by sfc-gh-swinkler — APPROVED** [review]

**Comment by sfc-gh-swinkler** [comment]
@astanfel thank you for your contribution!

**Pull request merged**